src/outputs/output_utils.hpp

```diff
@@ -200,6 +203,59 @@ struct AllSwarmInfo {
                bool is_restart);
 };
 
+template <typename T, typename Function_t>
+std::vector<T> FlattenBlockInfo(Mesh *pm, int shape, Function_t f) {
+  const int num_blocks_local = static_cast<int>(pm->block_list.size());
+  std::vector<T> data(shape * num_blocks_local);
+  int i = 0;
+  for (auto &pmb : pm->block_list) {
+    f(pmb.get(), data, i);
+  }
+  return data;
+}
+
+template <typename Data_t, typename idx_t, typename Function_t>
+void PackOrUnpackVar(MeshBlock *pmb, Variable<Real> *pvar, bool do_ghosts, idx_t &idx,
```

Comment:
Worth noting, @pgrete this is the one function I am likely to change down the road, if you wanted to use it. I plan to have this machinery return some shape information, and also permute it to support cells/edges/etc.

Reply:
No worries.
I'll keep an eye on this.
Also I might later also modify this function in order to allow for coarsening of the data (so overall I'm very much in favor of having this abstraction/functionality).